app/src/main/java/ch/hikemate/app/ui/auth/SignInScreen.kt

```diff
@@ -83,83 +85,93 @@ fun SignInScreen(
             Log.d("MainActivity", "addAccountLauncher result: $result")
           }
 
+  val errorMessageIdState = authViewModel.errorMessageId.collectAsState()
+  val loadingState = authViewModel.loading.collectAsState()
+
   // If the user is already signed in, navigate to the map screen
   LaunchedEffect(authViewModel.currentUser.collectAsState().value) {
     if (authViewModel.isUserLoggedIn()) {
       navigationActions.navigateTo(TopLevelDestinations.MAP)
     }
   }
 
-  Scaffold(
-      modifier = Modifier.fillMaxSize().testTag(Screen.AUTH),
-      content = { padding ->
-        Box(
-            modifier = Modifier.fillMaxSize(),
-        ) {
-          // I created a box to make the background image take a little more space
-          // than the screen size, so that the blur effect doesn't show white edges
+  AsyncStateHandler(
+      errorMessageIdState = errorMessageIdState,
+      actionContentDescriptionStringId = R.string.retry,
+      actionOnErrorAction = { navigationActions.navigateTo(Route.AUTH) },
```

Comment:
Warning: I believe this will result in a soft loop. Could we not clear the error to allow the user to retry instead?
Could you add a test about this too?

Reply:
Yep

Reply:
Thanks!